llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -5737,10 +5741,15 @@ LoopVectorizationCostModel::selectInterleaveCount(ElementCount VF,
   // the InterleaveCount as if vscale is '1', although if some information about
   // the vector is known (e.g. min vector size), we can make a better decision.
   if (BestKnownTC) {
-    MaxInterleaveCount =
-        std::min(*BestKnownTC / VF.getKnownMinValue(), MaxInterleaveCount);
-    // Make sure MaxInterleaveCount is greater than 0.
-    MaxInterleaveCount = std::max(1u, MaxInterleaveCount);
```

Comment:
It's interesting that previously we might have chosen IC to be a non-power-of-2 value!

Reply:
Yes. There existed at least one path where MaxInterleaveCount could have been a non-power-of-2 that may have contributed to IC being a non-power-of-2 later.